### PR TITLE
Added workflow for automating releases

### PR DESCRIPTION
**Pull request by GevinHasmitha**
Added workflow for creating new tag and publishing the release.

### CODE REVIEW

File: .github/workflows/cd-publishRelease.yml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+jobs:
+  release:
+    runs-on: ubuntu-latest
+
+    steps:
+     - name: Checkout code
+       uses: actions/checkout@v3
+       with:
+        ref: ${{ github.event.inputs.target-branch }}
+        token: ${{ secrets.TAG_TOKEN }}
```
Comment (ThishaniLucas) [inline .github/workflows/cd-publishRelease.yml:19]: Do we need these? Why are we taking the branch name here?

File: .github/workflows/cd-publishRelease.yml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+       
+     - name: Get current branch
```
Comment (ThishaniLucas) [inline .github/workflows/cd-publishRelease.yml:21]: We may not need this variable. We have the branch info as an input.

File: .github/workflows/cd-publishRelease.yml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+       run: |
+          branch=$(git rev-parse --abbrev-ref HEAD)
+          echo "branch=$branch" >> $GITHUB_ENV
+          echo "${{ github.workspace }}"
+
+     - name: Run ballerina build
+       uses: ballerina-platform/ballerina-action@2201.7.0
+       with:
+        args:
+          pack
```
Comment (ThishaniLucas) [inline .github/workflows/cd-publishRelease.yml:31]: This should be bal build. That's the command producing the jar file.

**Review by ThishaniLucas — CHANGES_REQUESTED** [review]

File: .github/workflows/cd-publishRelease.yml (hunk 1 above)
Comment (GevinHasmitha) [reply .github/workflows/cd-publishRelease.yml:19]: We checkout to the required branch and then we can run the bal build (from inside the branch) which will build the jar file for the contents of that branch

Comment (ThishaniLucas) [reply .github/workflows/cd-publishRelease.yml:19]: Check https://github.com/actions/checkout
When we select the branch from which this workflow should run, that'll be the ref. So need to specify this. Also we don't need the token to access this code.

File: .github/workflows/cd-publishRelease.yml [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+       run: |
+          jar_files=$(find "${{ github.workspace }}" -name "*.jar" -path "*/target/bin/*")
+          echo "JAR_FILES=$jar_files" >> $GITHUB_ENV
+
+     - name: JAR file Info
+       run: |
+          echo "Jar path: $JAR_FILES"
+          jar tf $JAR_FILES
+
+     - name: Get tag from Ballerina.toml and create tag name
```
Comment (ThishaniLucas) [inline .github/workflows/cd-publishRelease.yml:41]: We don't have to create a tag explicitly and push. Creating a release will create a tag.

File: .github/workflows/cd-publishRelease.yml (hunk 1 above)
Comment (GevinHasmitha) [reply .github/workflows/cd-publishRelease.yml:19]: Yes, I have included the branch selection in the ref but if i didn't include the access token, it gives an authentication error🤔

File: .github/workflows/cd-publishRelease.yml [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+              "tag_name": "'"$VERSION"'",
+              "target_commitish": "${{ github.event.inputs.target-branch }}",
+              "name": "'"$VERSION"'",
+              "body": "Releases",
+              "draft": false,
+              "prerelease": false,
+              "generate_release_notes": false
+            }'
+
+     - name: Make HTTP GET request
```
Comment (ThishaniLucas) [inline .github/workflows/cd-publishRelease.yml:67]:
```suggestion
     - name: Upload JAR to release
```

**Review by ThishaniLucas — APPROVED** [review]

**Pull request merged**